packages/vue/_src/hooks.ts

```diff
@@ -207,89 +207,76 @@ export function make<R, E, A>(self: Effect<R, E, FetchResponse<A>>) {
 }
 
 /**
- * Pass a function that returns an Effect, e.g from a client action.
+ * Pass a function that returns an Effect, e.g from a client action, or an Effect
  * Returns a tuple with state ref and execution function which reports errors as Toast.
  */
-export function useMutation<I, E, A>(self: (i: I) => Effect<ApiConfig | Http, E, A>) {
+export const useMutation: {
+  <I, E, A>(self: (i: I) => Effect<ApiConfig | Http, E, A>): readonly [
+    ComputedRef<{
+      loading: boolean
+      value: A | undefined
+      error: E | undefined
+    }>,
+    (i: I, abortSignal?: AbortSignal) => Promise<Left<E, A> | Right<E, A>>
+  ]
+  <E, A>(self: Effect<ApiConfig | Http, E, A>): readonly [
+    ComputedRef<{
+      loading: boolean
+      value: A | undefined
+      error: E | undefined
+    }>,
+    (abortSignal?: AbortSignal) => Promise<Left<E, A> | Right<E, A>>
+  ]
+} = <I, E, A>(self: ((i: I) => Effect<ApiConfig | Http, E, A>) | Effect<ApiConfig | Http, E, A>) => {
   const loading = ref(false)
   const error = ref<E>()
-  const value = ref<A>()
-  const handle = handleExit(loading, error, value)
-  const exec = (i: I, abortSignal?: AbortSignal) =>
+  const success = ref<A>()
+  const handle = handleExit(loading, error, success)
+
+  const exec = (...args: [i: I, abortSignal?: AbortSignal]) =>
```

Comment:
I think I would return two different functions, based on the typeof self, this way we only do the distinction on construction, not twice on every invocation.